vllm/v1/core/sched/scheduler.py

```diff
@@ -1326,11 +1326,31 @@ def finish_requests(
         waiting_requests_to_remove = []
         valid_requests = []
 
+        # this is only required only if we have a kv connector
+        should_force_abort = (
+            finished_status == RequestStatus.FINISHED_ABORTED
+            and self.get_kv_connector() is not None
+        )
+        forced_aborted_requests = []
```

Comment:
have you tried to set the timeout to a much lower value? if I understand correctly, this fix is freeing requests asap, which should be similar. Not sure if this is optimal in the case D is hogged as well